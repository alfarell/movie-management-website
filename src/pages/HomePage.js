import React from 'react';
import DisplayFilters from '../components/DisplayFilters';
import DisplayMovieList from '../components/DisplayMovieList';
import { Row, Col, Card, Typography } from 'antd';
import { FilterOutlined } from '@ant-design/icons';

const { Title } = Typography;
const HomePage = () => {
    return (
        <Row gutter={[10, 10]}>
            <Col xs={24} sm={24} md={24} lg={5} xl={4} xxl={4}>
                <Card title={<Title level={4} style={{ margin: 0 }}><FilterOutlined /> Discover</Title>}>
                    <DisplayFilters />
                </Card>
            </Col>
            <Col xs={24} sm={24} md={24} lg={19} xl={20} xxl={20}>
                <Card className='display-movie-list'>
                    <DisplayMovieList />
                </Card>
            </Col>
        </Row>
    );
};

export default HomePage;
