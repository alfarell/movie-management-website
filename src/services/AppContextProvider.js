import React, { createContext, useState, useEffect } from 'react';
import _ from 'loadsh';
import Axios from 'axios';


export const MovieContext = createContext();
export const UserContext = createContext();

const AppContextProvider = ({ children }) => {
    const [movieList, setMovieList] = useState([]);
    const [genreList, setGenreList] = useState([]);
    const [selectedGenre, setSelectedGenre] = useState(null);
    const [sortOption, setSortOption] = useState('popularity.desc')
    const [pagination, setPagination] = useState(1);


    useEffect(() => {
        Axios.get(`${process.env.REACT_APP_BASE_URL}/genre/list?api_key=${process.env.REACT_APP_BASE_API_KEY}&language=en-US`)
            .then(res => {
                setGenreList(res.data.genres);
            })
            .catch(err => {
                console.log(err);
            });
    }, [genreList]);

    useEffect(() => {
        const discovers = {
            genre: selectedGenre ? selectedGenre.id : '',
            sort: sortOption,
        };

        setPagination(1);
        fetchMovie(discovers);
    }, [selectedGenre, sortOption]);

    const fetchMovie = ({ genre, sort }) => {
        Axios.get(`${process.env.REACT_APP_BASE_URL}/discover/movie?api_key=${process.env.REACT_APP_BASE_API_KEY}&language=en-US&include_adult=false&include_video=false&page=${pagination}&with_genres=${genre}&sort_by=${sort}`)
            .then(res => {
                setMovieList([...res.data.results])
            })
            .catch(err => {
                console.log(err);
            });
    }

    const handleLoadMore = () => {
        setPagination(pagination + 1);

        const genre = selectedGenre ? selectedGenre.id : '';
        const sort = sortOption;
        let page = pagination + 1

        Axios.get(`${process.env.REACT_APP_BASE_URL}/discover/movie?api_key=${process.env.REACT_APP_BASE_API_KEY}&language=en-US&include_adult=false&include_video=false&page=${page}&with_genres=${genre}&sort_by=${sort}`)
            .then(res => {
                setMovieList([...movieList, ...res.data.results])
            })
            .catch(err => {
                console.log(err);
            });
    }

    const addFavoriteMovie = (favoritedMovie) => {
        let currentMovie = JSON.parse(localStorage.getItem('favorited'));
        let addFavorited;

        if (currentMovie) {
            let check = _.find(currentMovie, favoritedMovie);
            if (check) {
                addFavorited = currentMovie.filter(movie => {
                    return movie.id !== favoritedMovie.id
                });
            } else {
                addFavorited = [...currentMovie, favoritedMovie]
            }
        } else {
            addFavorited = [favoritedMovie];
        }

        localStorage.setItem('favorited', JSON.stringify(addFavorited));
    }

    return (
        <MovieContext.Provider value={{
            movieList,
            setMovieList,
            genreList,
            setGenreList,
            selectedGenre,
            setSelectedGenre,
            sortOption,
            setSortOption,
            pagination,
            setPagination,
            handleLoadMore,
            addFavoriteMovie
        }}>
            <UserContext.Provider value={{ test: 'test' }}>
                {children}
            </UserContext.Provider>
        </MovieContext.Provider>
    );
};

export default AppContextProvider;
