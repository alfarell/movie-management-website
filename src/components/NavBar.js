import React, { useContext } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { Layout, Typography, Row, Col, Menu } from 'antd';
import { PlayCircleOutlined, MenuOutlined } from '@ant-design/icons';
import { AppPages } from '../utils/AppPages';
import { MovieContext } from '../services/AppContextProvider';


const { Header } = Layout;
const { Title, Text } = Typography;

const NavBar = () => {
    const location = useLocation();
    const { listFavoriteMovie } = useContext(MovieContext);

    return (
        <Header className='nav-bar'>
            <Row justify='space-between' >
                <Col style={{ display: 'flex', alignItems: 'center' }}>
                    <Title style={{ margin: 0 }}>
                        <PlayCircleOutlined />
                        Movie
                    </Title>
                </Col>
                <Col xs={1} sm={9} md={8} lg={6} xl={5} xxl={4}>
                    <Menu
                        className='nav-menu'
                        mode='horizontal'
                        theme='light'
                        selectedKeys={[location.pathname]}
                        overflowedIndicator={<MenuOutlined style={{ fontSize: 16 }} />}
                    >
                        {AppPages.map(page => {
                            return (
                                <Menu.Item key={page.path} style={{ flex: 1, display: 'flex', justifyContent: 'center' }}>
                                    <Link to={page.path}>{page.name}</Link>
                                    {page.name === 'Favorites' ? <Text>({listFavoriteMovie.length})</Text> : ''}
                                </Menu.Item>
                            );
                        })}
                    </Menu>
                </Col>
            </Row>
        </Header>
    );
};

export default NavBar;
