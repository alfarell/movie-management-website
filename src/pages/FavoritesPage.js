import React, { useContext } from 'react';
import { Card, Row, Col } from 'antd';
import { MovieContext } from '../services/AppContextProvider';
import CardContent from '../components/CardContent';

const FavoritesPage = () => {
    const { favoritedMovie } = useContext(MovieContext);

    return (
        <Card>
            <Row>
                {favoritedMovie.map(movie => {
                    return (
                        <Col key={movie.id} xs={11} sm={7} md={6} lg={6} xl={4} xxl={3} >
                            <CardContent data={movie} />
                        </Col>
                    )
                })}
            </Row>
        </Card>
    );
};

export default FavoritesPage;
