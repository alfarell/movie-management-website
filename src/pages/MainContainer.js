import React, { Fragment } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Layout } from 'antd';
import NavBar from '../components/NavBar';
import HomePage from './HomePage';
import FavoritesPage from './FavoritesPage';
import MovieDetailPage from './MovieDetailPage';


const { Header, Content, Footer } = Layout;
const MainContainer = () => {
    return (
        <Router>
            <Header style={{ padding: 0 }}>
                <NavBar />
            </Header>
            <Content>
                <Switch>
                    <Route path='/' exact component={HomePage} />
                    <Route path='/favorites' component={FavoritesPage} />
                    <Route path='/movie/:id' component={MovieDetailPage} />
                </Switch>
            </Content>
            <Footer>Footer</Footer>
        </Router>
    );
};

export default MainContainer;
