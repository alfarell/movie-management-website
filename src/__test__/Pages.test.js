import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from '../pages/HomePage';
import MainContainer from '../pages/MainContainer';
import FavoritesPage from '../pages/FavoritesPage';
import AppContextProvider from '../services/AppContextProvider';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // deprecated
            removeListener: jest.fn(), // deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
            split: jest.fn()
        })),
    });
});

// test('render main container', () => {
//     const component = renderer.create(
//         <AppContextProvider>
//             <MemoryRouter initialEntries={['/favorites']}>
//                 <MainContainer />
//             </MemoryRouter>
//         </AppContextProvider>
//     );
//     const tree = component.toJSON();

//     expect(tree).toMatchSnapshot();
// });

test('render home page', () => {
    const component = renderer.create(
        <AppContextProvider>
            <HomePage />
        </AppContextProvider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('render favorite page', () => {
    const component = renderer.create(
        <AppContextProvider>
            <FavoritesPage />
        </AppContextProvider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
