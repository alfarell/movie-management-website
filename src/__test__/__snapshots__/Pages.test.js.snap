// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render favorite page 1`] = `
<div
  style={
    Object {
      "padding": "10px 20px",
    }
  }
>
  <div
    className="ant-card ant-card-bordered"
    style={
      Object {
        "minHeight": "75vh",
      }
    }
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Favorite Movie
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-row ant-row-center"
        style={
          Object {
            "marginBottom": 2.5,
            "marginLeft": -2.5,
            "marginRight": -2.5,
            "marginTop": -2.5,
          }
        }
      >
        <div
          className="ant-empty"
        >
          <div
            className="ant-empty-image"
          >
            <svg
              className="ant-empty-img-default"
              height="152"
              viewBox="0 0 184 152"
              width="184"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  transform="translate(24 31.67)"
                >
                  <ellipse
                    className="ant-empty-img-default-ellipse"
                    cx="67.797"
                    cy="106.89"
                    rx="67.797"
                    ry="12.668"
                  />
                  <path
                    className="ant-empty-img-default-path-1"
                    d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
                  />
                  <path
                    className="ant-empty-img-default-path-2"
                    d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                    transform="translate(13.56)"
                  />
                  <path
                    className="ant-empty-img-default-path-3"
                    d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
                  />
                  <path
                    className="ant-empty-img-default-path-4"
                    d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
                  />
                </g>
                <path
                  className="ant-empty-img-default-path-5"
                  d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
                />
                <g
                  className="ant-empty-img-default-g"
                  transform="translate(149.65 15.383)"
                >
                  <ellipse
                    cx="20.654"
                    cy="3.167"
                    rx="2.849"
                    ry="2.815"
                  />
                  <path
                    d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <p
            className="ant-empty-description"
          >
            No Favorite Movie
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`render home page 1`] = `
<div
  className="ant-row"
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": -5,
      "marginRight": -5,
      "marginTop": -5,
      "padding": "10px 20px",
    }
  }
>
  <div
    className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-5 ant-col-xl-4 ant-col-xxl-4"
    style={
      Object {
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
  >
    <div
      className="ant-card ant-card-bordered"
    >
      <div
        className="ant-card-head"
        style={Object {}}
      >
        <div
          className="ant-card-head-wrapper"
        >
          <div
            className="ant-card-head-title"
          >
            <h4
              className="ant-typography"
              style={
                Object {
                  "WebkitLineClamp": null,
                  "margin": 0,
                }
              }
            >
              <span
                aria-label="filter"
                className="anticon anticon-filter"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="filter"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880.1 154H143.9c-24.5 0-39.8 26.7-27.5 48L349 597.4V838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V597.4L907.7 202c12.2-21.3-3.1-48-27.6-48zM603.4 798H420.6V642h182.9v156zm9.6-236.6l-9.5 16.6h-183l-9.5-16.6L212.7 226h598.6L613 561.4z"
                  />
                </svg>
              </span>
               Discover
            </h4>
          </div>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <ul
          className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
          onMouseEnter={[Function]}
          onTransitionEnd={[Function]}
          role="menu"
          style={Object {}}
        >
          <div />
          <li
            className="ant-menu-submenu ant-menu-submenu-inline"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-menu-submenu-title"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="button"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
              title="Sort By"
            >
              Sort By
              <i
                className="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            className="ant-menu-submenu ant-menu-submenu-inline ant-menu-submenu-disabled"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-menu-submenu-title"
              role="button"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <span>
                Select Genre
              </span>
              <i
                className="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-19 ant-col-xl-20 ant-col-xxl-20"
    style={
      Object {
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
  >
    <div
      className="ant-card display-movie-list ant-card-bordered"
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-row ant-row-center"
          style={
            Object {
              "marginBottom": 5,
              "marginLeft": -5,
              "marginRight": -5,
              "marginTop": -5,
            }
          }
        >
          <div
            className="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text"
          >
            <span
              className="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
            </span>
            <div
              className="ant-spin-text"
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
